src/router.ts

```diff
@@ -48,6 +48,7 @@ export const router = kea<routerType>([
       hashParams,
       initial: initial || false,
     }),
+    clearUnloadInterceptors: true, // Useful for "save" events where you don't want to wait for a re-render before a push
```

Comment:
I guess this is fine, and I guess essential for the hook to work well... but it's also a workaround that requires a lot of context to fully understand as a user of the library.

Normally I try to "teach" Kea as a separate data layer, and the more processing happens in that layer, the better. If this is followed well, then the knowledge of "should the beforeunload be active or not" is also there within Kea.

Thus, two thoughts, one easier than the other:
- could there be a way to activate or deactivate a `beforeUnload` from within a logic itself
- could we pass a selector to the hook instead of a value, so it would always double check if it needs to actually confirm.

The first option might look something like:
```ts
import { kea } from 'kea'
import { beforeUnload } from 'kea-router'

const logic = kea<logicType>({
  forms({
    loginForm: {
      defaults: { user: '', pass: '' },
      submit: () => throw new Error('access denied')
    }
  }),
  beforeUnload(({ values }) => ({
    enabled: () => !values.loginForm.user,
    message: "Come on now, you've come so far...",
  }))
})
```

The second option might look like

```ts
function Component {
  const builtLogic = useMountedLogic(logic)
  useUnloadConfirmation(builtLogic.selectors.unconfirmMessage)
  // ...
}
```

Reply:
This actually works way better. Update coming